nn/conformer.py

```diff
@@ -178,40 +179,27 @@ class ConformerEncoder(nn.Module):
   """
 
   def __init__(
-        self, encoder_layer: nn.Module, num_blocks: int, conv_kernel_size: int = 32,
-        activation_ff: Callable[[nn.LayerRef], nn.LayerRef] = nn.swish, dim_ff: int = 512, dropout: float = 0.1,
-        att_dropout: float = 0.1, enc_key_dim: int = 256, num_heads: int = 4,
-        batch_norm_opts: Optional[Dict[str, Any]] = None):
+        self, encoder_layer: Union[ConformerEncoderLayer, Any], num_blocks: int):
     """
     :param encoder_layer:
     :param num_blocks:
-    :param conv_kernel_size:
-    :param ff_act:
-    :param ff_dim:
-    :param dropout:
-    :param att_dropout:
-    :param enc_key_dim:
-    :param att_n_heads:
     """
     super().__init__()
 
-    self.dropout = dropout
+    self.dropout = getattr(encoder_layer, 'dropout')
```

Comment:
I am not sure what is the best way to handle these two attributes. We can pass the dropout as parameter to this class instead so we can set a different dropout rate for the conv subsampling layer. I followed ESPNet recipe for that.

Reply:
So in the Conformer architecture, there is a conv subsampling layer, linear layer, and dropout layer before the conformer blocks. These layers are part of the ConfromerEncoder. The question is if we should decouple this.

Reply:
Good question. Definitely you should not use `getattr` here. You might also restrict the type of `encoder_layer` to `ConformerEncoderLayer`. Then you could just do:
```suggestion
    self.dropout = encoder_layer.dropout
```
Anyone who wants to use another custom encoder layer type can simply use his own class. It doesn't make sense that `ConformerEncoder` is so generic here.